crates/matrix-sdk/tests/integration/room/joined.rs

```diff
@@ -713,3 +715,51 @@ async fn test_call_notifications_notify_for_rooms() {
 
     room.send_call_notification_if_needed().await.unwrap();
 }
+
+#[async_test]
+async fn test_start_live_location_share_for_room() {
+    let (client, server) = logged_in_client_with_server().await;
+
+    Mock::given(method("PUT"))
+        .and(path_regex(r"^/_matrix/client/r0/rooms/.*/state/org.matrix.msc3672.beacon_info/.*"))
+        .and(header("authorization", "Bearer 1234"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(&*test_json::EVENT_ID))
+        .mount(&server)
+        .await;
+
+    mock_sync(&server, &*test_json::SYNC, None).await;
+
+    let sync_settings = SyncSettings::new().timeout(Duration::from_millis(3000));
+
+    let _response = client.sync_once(sync_settings).await.unwrap();
+
+    let room = client.get_room(&DEFAULT_TEST_ROOM_ID).unwrap();
+
+    let response = room.start_live_location_share(3000, None).await.unwrap();
+
+    assert_eq!(event_id!("$h29iv0s8:example.com"), response.event_id);
```

Comment:
I think it would be good to assert here that the beacon event was actually added to the room state.

Reply:
updated this test to be more comprehensive, let me know if its the right direction!